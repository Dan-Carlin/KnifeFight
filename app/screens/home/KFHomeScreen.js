/*
KFHomeScreen - View for the home screen of the app.
*/

// External libraries
import React, { useEffect } from "react";
import { View, Image } from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { useDispatch } from "react-redux";

// Assets
import { FGLogo, KFLogo } from "../../assets/logos";
import Gradient from "../../assets/backgrounds/split_gradient_green.svg";
import {
  SettingsGraphic,
  StartGameGraphic,
} from "../../assets/buttons/navigation";

// Components
import OpacityButton from "../../components/OpacityButton";
import Screen from "../../components/Screen";
import Text from "../../components/Text";

// Resources
import routes from "../../navigation/routes";
import { setBaseHp, setEnableAudio, setShowPopup } from "../../redux/actions";
import sounds from "../../assets/sounds/sounds";
import strings from "../../config/strings";
import styles from "./KFHomeStyles";
import useAudioController from "../../hooks/useAudioController";

const background = require("../../assets/backgrounds/kf_background_xxxhdpi.png");

function KFHomeScreen({ navigation }) {
  const { enableSounds } = useAudioController();

  const sgLogo = require("../../assets/logos/sg_logo_light.png");
  const dispatch = useDispatch();

  const LoadSettings = async () => {
    try {
      const settingsValue = await AsyncStorage.getItem("@settings");
      const settings = JSON.parse(settingsValue);

      if (settingsValue !== null) {
        dispatch(setBaseHp(settings.baseHp));
        dispatch(setEnableAudio(settings.enableAudio));
        dispatch(setShowPopup(settings.showPopup));

        enableSounds(settings.enableAudio);
      }
    } catch (e) {
      console.log(e);
    }
  };

  useEffect(() => {
    LoadSettings();
  }, []);

  return (
    <Screen style={styles.screenContainer} background={background}>
      <View style={styles.logoContainer}>
        <KFLogo style={styles.logo} />
        <View style={styles.settingsButton}>
          <OpacityButton
            Graphic={SettingsGraphic}
            sound={sounds.NEXT_CONFIRM}
            onPress={() => {
              navigation.navigate(routes.SETTINGS);
            }}
          />
        </View>
      </View>
      <View style={styles.buttonContainer}>
        <Gradient style={styles.gradient} />
        <View style={styles.button}>
          <OpacityButton
            Graphic={StartGameGraphic}
            sound={sounds.START_END_TURN}
            onPress={() => {
              setTimeout(() => {
                navigation.navigate(routes.SETUP_STEP_ONE);
              }, 100);
            }}
          />
        </View>
      </View>
      <View style={styles.bottomContainer}>
        <View style={styles.bottomContents}>
          <Text style={styles.disclaimer}>
            <Text style={styles.disclaimerBold}>
              {strings.home_disclaimer_title}
            </Text>
            <Text style={styles.disclaimer}>
              {strings.home_disclaimer_text_a}
            </Text>
            <Text style={styles.disclaimerItalic}>{strings.app_name}</Text>
            <Text style={styles.disclaimer}>
              {strings.home_disclaimer_text_b}
            </Text>
          </Text>
          <View style={styles.companyLogos}>
            <Image style={styles.sgLogo} source={sgLogo} />
            <FGLogo style={styles.fgLogo} />
          </View>
        </View>
      </View>
    </Screen>
  );
}

export default KFHomeScreen;
