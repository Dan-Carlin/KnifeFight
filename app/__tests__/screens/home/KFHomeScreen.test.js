import React from "react";
import renderer from "react-test-renderer";

import KFHomeScreen from "../../../screens/home/KFHomeScreen";

describe("Home screen tests", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<KFHomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
