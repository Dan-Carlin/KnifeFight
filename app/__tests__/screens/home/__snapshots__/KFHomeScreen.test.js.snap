// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen tests renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "paddingTop": 20,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.6,
        }
      }
    >
      <SvgMock
        style={
          Object {
            "height": "48.2%",
            "marginTop": 10,
            "width": "85%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.18,
          "justifyContent": "center",
        }
      }
    >
      <SvgMock
        style={
          Object {
            "position": "absolute",
          }
        }
      />
      <View
        style={
          Object {
            "height": "85%",
            "paddingBottom": 14,
            "paddingHorizontal": 40,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.22,
        }
      }
    >
      <View
        style={
          Object {
            "height": "100%",
            "width": "55%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "width": "100%",
              },
              Object {
                "fontSize": 12,
                "marginTop": 15,
                "textAlign": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontSize": 12,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            DISCLAIMER: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontSize": 12,
                  "marginTop": 15,
                  "textAlign": "center",
                },
              ]
            }
          >
            This is a companion app for the 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontSize": 12,
                  "fontStyle": "italic",
                },
              ]
            }
          >
            Knife Fight
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontSize": 12,
                  "marginTop": 15,
                  "textAlign": "center",
                },
              ]
            }
          >
             board game, not a standalone game.
          </Text>
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "flex": 1,
                "height": "100%",
                "marginEnd": 30,
                "opacity": 0.5,
                "resizeMode": "contain",
              }
            }
          />
          <SvgMock
            style={
              Object {
                "height": "100%",
                "opacity": 0.5,
                "width": "30%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
